src/test/kotlin/org/opensearch/replication/task/index/IndexReplicationTaskTests.kt

```diff
@@ -0,0 +1,204 @@
+import org.opensearch.action.admin.indices.settings.get.GetSettingsAction
+import org.opensearch.action.admin.indices.settings.get.GetSettingsResponse
+import org.opensearch.action.admin.indices.settings.put.UpdateSettingsAction
+import org.opensearch.action.support.master.AcknowledgedResponse
+import org.opensearch.cluster.ClusterState
+import org.opensearch.cluster.ClusterStateObserver
+import org.opensearch.cluster.RestoreInProgress
+import org.opensearch.cluster.metadata.IndexMetadata
+import org.opensearch.cluster.metadata.Metadata
+import org.opensearch.cluster.routing.RoutingTable
+import org.opensearch.common.settings.Settings
+import org.opensearch.common.settings.SettingsModule
+import org.opensearch.common.unit.TimeValue
+import org.opensearch.index.Index
+import org.opensearch.index.shard.ShardId
+import org.opensearch.persistent.PersistentTaskParams
+import org.opensearch.persistent.PersistentTasksCustomMetadata
+import org.opensearch.persistent.PersistentTasksService
+import org.opensearch.snapshots.Snapshot
+import org.opensearch.snapshots.SnapshotId
+import org.opensearch.tasks.TaskId.EMPTY_TASK_ID
+import org.opensearch.tasks.TaskManager
+import org.opensearch.test.ClusterServiceUtils
+import org.opensearch.test.ClusterServiceUtils.setState
+import org.opensearch.test.OpenSearchTestCase
+import org.opensearch.test.OpenSearchTestCase.assertBusy
+import org.opensearch.threadpool.TestThreadPool
+import org.junit.Test
+import org.mockito.Mockito
+import org.opensearch.common.xcontent.NamedXContentRegistry
+import org.opensearch.replication.metadata.ReplicationOverallState
+import org.opensearch.replication.metadata.store.ReplicationMetadataStore
+import org.opensearch.replication.metadata.store.ReplicationMetadataStore.Companion.REPLICATION_CONFIG_SYSTEM_INDEX
+import org.opensearch.replication.metadata.store.ReplicationStoreMetadataType
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+@ThreadLeakScope(ThreadLeakScope.Scope.NONE)
+class IndexReplicationTaskTests : OpenSearchTestCase()  {
+
+    companion object {
+        var currentTaskState :IndexReplicationState = InitialState
+        var stateChanges :Int = 0
+        var restoreNotNull = false
+
+        var followerIndex = "follower-index"
+        var connectionName = "leader-cluster"
+        var remoteCluster = "remote-cluster"
+    }
+
+    var threadPool = TestThreadPool("ReplicationPluginTest")
+    var clusterService  = ClusterServiceUtils.createClusterService(threadPool)
+
+    @Test
+    fun testExecute() = runBlocking {
+        val replicationTask: IndexReplicationTask = spy(createIndexReplicationTask())
+        var taskManager = Mockito.mock(TaskManager::class.java)
+        replicationTask.setPersistent(taskManager)
+        var rc = ReplicationContext(followerIndex)
+        var rm = ReplicationMetadata(connectionName, ReplicationStoreMetadataType.INDEX.name, ReplicationOverallState.RUNNING.name, "reason", rc, rc, Settings.EMPTY)
```

Comment:
Minor: We can probably move the logic to create `ReplicationMetadata`, `ClusterState`, etc objects  to a helper utility